prow/README.md

```diff
@@ -32,6 +32,10 @@ state and no claims of backwards compatibility are made for any external API.
    plugin configuration from the main `ConfigMap`. For more context, please see
    [this pull request.](https://github.com/kubernetes/test-infra/pull/4213)
 
+## Turning up a new prow cluster
```

Comment:
Can we make this less jargony? Cluster, turn up don't scream "I'm a newbie, here's where I want to go". How about "Getting started guide" or something else easier to understand?

Reply:
Done.